### PR TITLE
Unit test for PUBDEV-3401 & fixes in R function h2o.parseSetup

### PR DESCRIPTION
**Pull request by michalkurka**
This pull request is closely related to PR #211 

Adding @ledell to review because of changes in h2o.parseSetup API.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/214)

### CODE REVIEW

**Comment by michalkurka** [comment]
Updated the test to use existing data in smalldata (I will need to add just one file to smalldata). This doesn't rely on the data I got from Darren Cook.

File: h2o-r/tests/testdir_jira/runit_pubdev_3401.R [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+test.pubdev_3401 <- function() {
+    files <- c(locate("smalldata/jira/adult.gz"), locate("smalldata/jira/pubdev_3401-adult-1000.gz"))
+
+    data.raw <- h2o.importFile(files, destination_frame = "pubdev3401.raw", parse = FALSE)
+    data.parsed <- h2o.parseRaw(data.raw, destination_frame = "pubdev3401.parsed",
+                                parse_type = "CSV", chunk_size = 105000, header = FALSE)
+
+    nr <- nrow(data.parsed)
```
Comment (michalkurka) [inline h2o-r/tests/testdir_jira/runit_pubdev_3401.R]: good catch!

**Comment by arnocandel** [comment]
@ledell are you ok with this? I'd merge this now.

**Comment by mmalohlava** [comment]
Thx Michal! Let's merge it when it finishes all checks

**Comment by abal5** [comment]
@h2o-ops please re-build

**Pull request merged**